src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -0,0 +1,762 @@
+        self.worker_thread.join()
+
+    def _do_correlation(self):
+        """Run the correlation between the FIB and FM images."""
+        # Modify the input data to match the required format to run 3DCT
+        fm_das = [stream.raw[0] for stream in self.correlation_target.fm_streams]
+        fm_image = _convert_das_to_numpy_stack(fm_das)
+        fib_da = self.correlation_target.fib_stream.raw[0]
+        fib_coords = []
+        fm_coords = []
+        poi_coords = []
+        path = self._tab_data_model.main.project_path.value
+        for fib_coord in self.correlation_target.fib_fiducials:
+            fib_coord = self.correlation_target.fib_stream.getPixelCoordinates(fib_coord.coordinates.value[0:2], check_bbox=False)
+            fib_coords.append(fib_coord)
+        fib_coords = numpy.array(fib_coords, dtype=numpy.float32)
+        for fm_coord in self.correlation_target.fm_fiducials:
+            fm_coord_2d = self.correlation_target.fm_streams[0].getPixelCoordinates(fm_coord.coordinates.value[0:2], check_bbox=False)
+            fm_coords.append([fm_coord_2d[0], fm_coord_2d[1], fm_coord.coordinates.value[2]])
+        fm_coords = numpy.array(fm_coords, dtype=numpy.float32)
+        poi_coord = self.correlation_target.fm_pois[0]
+        poi_coord_2d = self.correlation_target.fm_streams[0].getPixelCoordinates(poi_coord.coordinates.value[0:2], check_bbox=False)
+        poi_coords.append([poi_coord_2d[0], poi_coord_2d[1], poi_coord.coordinates.value[2]])
+        poi_coords = numpy.array(poi_coords, dtype=numpy.float32)
+        # Run the correlation
+        self.correlation_target.correlation_result = run_tdct_correlation(fib_coords=fib_coords, fm_coords=fm_coords,
+                                                                          poi_coords=poi_coords,
+                                                                          fib_image=fib_da, fm_image=fm_image,
+                                                                          path=path)
+        # Update the output pameters of the correlation
+        self._tab_data_model.projected_points = []
+        self.correlation_target.fib_projected_fiducials = []
+        points = self.correlation_target.correlation_result['output']['error']['reprojected_3d']
+        for n, i in enumerate(points[0]):
+            p_pos = self.correlation_target.fib_stream.getPhysicalCoordinates((points[0][n], points[1][n]))
+            target = Target(x=p_pos[0], y=p_pos[1], z=0, name="PP" + str(n + 1), type="ProjectedPoints", index=n + 1,
+                            fm_focus_position=0)
+            self._tab_data_model.projected_points.append(target)
+            self.correlation_target.fib_projected_fiducials.append(target)
+
+        # Convert from pixel to physical coordinates
+        projected_poi = (self.correlation_target.fib_stream.getPhysicalCoordinates
+                         (self.correlation_target.correlation_result["output"]["poi"][0]["image_px"]))
```

Comment:
Don't put the call on a separate line from the function, it makes it unclear that it's a function.
Better write on two steps:
```suggestion
        projected_poi_px = self.correlation_target.correlation_result["output"]["poi"][0]["image_px"]
        projected_poi = self.correlation_target.fib_stream.getPhysicalCoordinates(projected_poi_px)
```